smarts/core/road_map.py

```diff
@@ -625,6 +625,9 @@ def __eq__(self, other) -> bool:
             """Required for set usage; derived classes may override this."""
             return self.__class__ == other.__class__ and hash(self) == hash(other)
 
+        def _add_road(self, road: RoadMap.Road):
```

Comment:
Should we have an underscored method to be implemented by the child?

Reply:
The issue here was that the children were expected to have it so I had to require this to be implemented or any further implementation will break. Perhaps this should only be necessary in the `RouteWithCache` class rather than the base `RoadMap.Route`.

Reply:
On second look, it seems that all implementations of `RoadMap` assume this method exists on `Route`.